File: frontend/services/dashboardService.js
"use client";

import { apiFetch } from "./api";

export const getDashboardSummary = () => apiFetch("/dashboard/summary");
export const getDailyMetrics = (days = 7) => apiFetch(`/dashboard/daily-metrics?days=${days}`);
export const getAppointmentTrends = (days = 30) => apiFetch(`/dashboard/appointments/trends?days=${days}`);
export const getRevenueTrends = (days = 30) => apiFetch(`/dashboard/revenue/trends?days=${days}`);
export const getDiseaseTrends = () => apiFetch("/dashboard/disease-trends");
export const getTopDoctors = (limit = 5) => apiFetch(`/dashboard/top-doctors?limit=${limit}`);
export const getAppointmentStats = () => apiFetch("/dashboard/appointments");
export const getUserStats = () => apiFetch("/dashboard/users");
export const getRevenueStats = () => apiFetch("/dashboard/revenue");
export const getCompletionRate = () => apiFetch("/dashboard/completion-rate");
export const getDoctorPerformance = (doctorId) => apiFetch(`/dashboard/doctors/${doctorId}/performance`);
export const getDoctorAppointmentStats = (doctorId) => apiFetch(`/dashboard/doctors/${doctorId}/appointments`);

File: backend/src/main/java/com/hospital/backend/service/DashboardService.java
package com.hospital.backend.service;

import com.hospital.backend.dto.*;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

public interface DashboardService {

    // Overall dashboard
    DashboardSummaryDTO getDashboardSummary(String userId);

    // User statistics
    UserStatsDTO getUserStats();

    // Appointment statistics
    AppointmentStatsDTO getAppointmentStats();

    // Revenue statistics
    RevenueStatsDTO getRevenueStats();

    // Daily metrics for last N days
    List<DailyMetricsDTO> getDailyMetrics(int days);

    // Top performing doctors
    List<DoctorPerformanceDTO> getTopDoctors(int limit);

    // Appointments for specific doctor
    AppointmentStatsDTO getDoctorAppointmentStats(String doctorId);

    // Doctor performance metrics
    DoctorPerformanceDTO getDoctorPerformance(String doctorId);

    // Appointment trends (count by day for last N days)
    List<DailyMetricsDTO> getAppointmentTrends(int days);

    // Revenue trends
    List<DailyMetricsDTO> getRevenueTrends(int days);

    // Get appointments between dates
    long getAppointmentsCount(LocalDate startDate, LocalDate endDate);

    // Get completed appointments percentage
    double getCompletionRate();

    // Get disease trends (count by diagnosis)
    Map<String, Long> getDiseaseTrends();
}
